src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -30,13 +34,13 @@
      * Looks at the definition of {@link AbstractProject} and list up the related github repositories,
      * then puts them into the collection.
      */
-    public abstract void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result);
+    public abstract void parseAssociatedNames(Job<?,?> job, Collection<GitHubRepositoryName> result);
```

Comment:
This will be binary- and source-incompatible for plugins implementing this extension point. I cannot find any, but still…safer to add a new overload, deprecate this one, and use delegation and `Util.isOverridden` as needed to make it compatible.